### ISSUE TITLE
Pension List+Loop: Focus improperly managed at end of list and loop

### ISSUE THREAD

**Issue by kristinoletmuskat** [department-of-veterans-affairs/va.gov-team#105391]
## Brief description
As part of our list and loop enhancements, we submitted an [ADE ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/104338) for a11y review. This is one of the findings to investigate and address before we release the component live.

## Problem - Focus improperly managed at end of list and loop

**Page**: Add/edit/review a previous service name, Add/edit/review a VA medical center, Add/edit/review a federal medical center, Add/edit/review a dependent child
**Device**: MacOS
**Expected severity level**: 2
**Expected experience standard**: User experiences predictable interactions with components and patterns

**Description**: 
At all steps of the list and loop flow, focus should be managed to the H3 when moving between screens. For example, in the [mock form patterns](https://staging.va.gov/mock-form-patterns/treatment-records) implementation, “Name and address of treatment facility” gets focus after pressing Continue on the “Treatment records” page. ([Clip](https://us02web.zoom.us/clips/share/ELs-JGOqSwudj30M6PB0TA))
This isn’t the cast on 527EZ - focus is going to the progress bar instead. This makes it less clear for screen reader users, who might TAB and miss the heading entirely. ([Clip](https://us02web.zoom.us/clips/share/fNJw-hdKTAyS4KB5oG0xLA))

**How to recreate**: 
1. Turn on a screen reader
2. Choose “Yes, I have a previous name to report” and press continue
3. Note where focus is managed - to the progress bar

**Recommendation**: Manage focus to the H3 to match the mock form patterns implementation.
